### PR TITLE
catkin

### PR DESCRIPTION
**Pull request by andre-nguyen**
I'll be honest, we did this in a total rush and I suspect some lines are useless. We also removed a lot of variables which didn't pertain to our platform.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -40,19 +28,32 @@ else()
...
+  src/DebugImage.cpp
+  src/Geometry.cpp 
+  src/GrayModel.cpp
+  src/MathUtil.cpp 
+  src/Refine.cpp
+  src/TagDetector.cpp 
+  src/TagFamily.cpp
+  src/TagFamilies.cpp
+  src/UnionFindSimple.cpp
+  ${apriltags_swatbotics_headers}
```
Comment (andre-nguyen) [inline CMakeLists.txt]: I don't think we actually need to include the headers in the library declaration.

File: package.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<package format="2">
+  <name>apriltags_swatbotics</name>
+  <version>0.1.0</version>
+  <description>Swatbotics april tag library</description>
+  <author>Swatbotics</author>
+  <maintainer email="andre-phu-van.nguyen@polymtl.ca">Andre Phu-Van Nguyen</maintainer>
```
Comment (andre-nguyen) [inline package.xml]: This line should be changed I guess.

**Comment by andre-nguyen** [comment]
@psigen I doubt you would want to merge this as is since it's pretty messy, but the package works.

On the apriltags ros package side all you have to do is add 

```
find_package(catkin REQUIRED COMPONENTS
  cv_bridge
  std_msgs
  sensor_msgs
  image_transport
  roscpp
  cmake_modules
  message_generation
  geometry_msgs
  tf
  apriltags_swatbotics
)
```

and 

```
<depend>apriltags_swatbotics</depend>
```

to your package.xml (assuming you convert it to the new catkin tools format).

**Comment by mkoval** [comment]
I am working on adding a REP-136 compliant `package.xml` file. This avoids making this package depend on Catkin. I suggest we hold off merging this pull request and do that instead.

**Pull request closed without merging**